### ISSUE TITLE
Exception joining weird channel

### ISSUE THREAD

**Issue by greboid** [DMDirc/Parser#161]
Trying to join `#/%=~_|` causes the below exception
```
Parser error occurred: Exception in Processor. [ProcessJoin]: Conversion = '='
    Last line received: :<hostname> JOIN #/%=~_|
----
Exception: java.util.UnknownFormatConversionException: Conversion = '='
Exception Stack trace:
    java.util.UnknownFormatConversionException: Conversion = '='
    	at java.util.Formatter.checkText(Formatter.java:2579)
    	at java.util.Formatter.parse(Formatter.java:2565)
    	at java.util.Formatter.format(Formatter.java:2501)
    	at java.util.Formatter.format(Formatter.java:2455)
    	at java.lang.String.format(String.java:2940)
    	at com.dmdirc.parser.irc.IRCParser.callDebugInfo(IRCParser.java:588)
    	at com.dmdirc.parser.irc.processors.IRCProcessor.callDebugInfo(IRCProcessor.java:85)
    	at com.dmdirc.parser.irc.processors.ProcessJoin.process(ProcessJoin.java:189)
    	at com.dmdirc.parser.irc.ProcessingManager.process(ProcessingManager.java:141)
    	at com.dmdirc.parser.irc.IRCParser.processLine(IRCParser.java:1206)
    	at com.dmdirc.parser.irc.IRCParser.run(IRCParser.java:877)
    	at java.lang.Thread.run(Thread.java:748)
----
```

**Comment by csmith** [comment]
So this happens because callDebugInfo takes formatting args, and ProcessJoin is just concatting stuff instead of using proper formatting args.

When it gets passed to String.format the %= gets treated as a formatting string, fails dismally, and then summons some kind of eldritch horror to take a bite out of the parser's state.

**Comment by ShaneMcC** [comment]
Fixed by #162
